src/dashboardGenerator.js

```diff
@@ -0,0 +1,359 @@
+'use strict';
+
+
+const ANNOTATION_ELITE_COLOUR = '#98df8a';
+const ANNOTATION_HIGH_COLOUR = '#ffbb78';
+const ANNOTATION_MEDIUM_COLOUR = '#d62728';
+const MEAN_COLOUR = '#2ca02c';
+
+const WIDGET_HEIGHT = 6;
+const WIDGET_WIDTH = 8;
+
+
+// Create a dashboard containing multiple widgets, each based from a consistent template
+// The properties here map the metrics to a readable label
+// unit conversion is used to convert from seconds to a more meaningful unit, based on the metric.
+const MINUTES = {
+  unit: 60,
+  label: 'minutes'
+};
+
+const HOURS = {
+  unit: 60 * 60,
+  label: 'hours'
+};
+
+const DAYS = {
+  unit: 60 * 60 * 24,
+  label: 'days'
+};
+
+const ONE_DAYS = 60 * 60 * 24;
+const SEVEN_DAYS = 60 * 60 * 24 * 7;
+
+const applyLimits = (state) => {
+  // See: https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/cloudwatch_limits.html
+  const maxMetricsPerWidget = 100;
+  const maxMetricsPerServiceInAWidget = 2;
+  const maxServices = Math.floor(maxMetricsPerWidget / maxMetricsPerServiceInAWidget);
+
+  // For each service there will be three metrics (MTTR, MTBF, MTTF)
+  const maxSupportedMetrics = maxServices * 3;
+
+  if (state.metrics.length > maxSupportedMetrics) {
+    console.warn(`Maximum of ${maxSupportedMetrics} metrics are allowed in a single dashboard. Some metrics will not be reported.`);
+    state.metrics = state.metrics.sort((a, b) => (a.serviceName > b.serviceName) ? 1 : -1);
+    state.metrics = state.metrics.slice(0, maxSupportedMetrics);
```

Comment:
If the metrics have been truncated, then a warning message should be added to a text widget in dashboard itself.

Either a separate text widget altogether, or append to each of the existing text widgets.

Reply:
Done